rllib/offline/offline_env_runner.py

```diff
@@ -164,6 +164,8 @@ def sample(
                 # TODO (simon): Find a better way to write these data.
                 self._samples = self._samples[self.output_max_rows_per_file :]
                 samples_ds = ray.data.from_items(samples_to_write)
+                # Delete reference to free memory.
+                del samples_to_write
```

Comment:
Why do we need to do this? Python's GC should take care of it when the variable goes out of scope, no?
It also keeps all the data anyways b/c it's just a slice from the `self._samples` ndarray.